src/conversion.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  AptosConfig,
+  Network,
+  InputGenerateTransactionPayloadData,
+  InputEntryFunctionDataWithRemoteABI,
+  TypeTag,
+  Deserializer,
+  Serializable as SerializableV2,
+  generateTransactionPayload
+} from "@aptos-labs/ts-sdk";
+import { areBCSArguments, NetworkInfo } from '@aptos-labs/wallet-adapter-core';
+import { BCS, TxnBuilderTypes, Types } from "aptos";
+
+export type SerializableV1 = { serialize(serializer: BCS.Serializer): void };
+export type DeserializableV2Class<T> = { deserialize(deserializer: Deserializer): T };
+export type DeserializableV1Class<T> = { deserialize(deserializer: BCS.Deserializer): T };
+
+export function convertV1toV2<TV1 extends SerializableV1, TV2>(src: TV1, dst: DeserializableV2Class<TV2>) {
+  const serializedBytes = BCS.bcsToBytes(src);
+  const deserializerV2 = new Deserializer(serializedBytes);
+  return dst.deserialize(deserializerV2);
+}
+
+export function convertV2toV1<TV1, TV2 extends SerializableV2>(src: TV2, dst: DeserializableV1Class<TV1>) {
+  const serializedBytes = src.bcsToBytes();
+  const deserializerV1 = new BCS.Deserializer(serializedBytes);
+  return dst.deserialize(deserializerV1);
+}
+
+// old => new
+export function convertNetwork(networkInfo: NetworkInfo | null): Network {
```

Comment:
the adapter holds a conversion layer with some same functionality - we might want to keep all conversions there
https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/packages/wallet-adapter-core/src/conversion.ts

Reply:
See my comment https://github.com/aptos-labs/petra-plugin-wallet-adapter/pull/15#issuecomment-1906751125